lib/assessment.js

```diff
@@ -29,9 +39,17 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  ary = [];
+  ary.length = length;
+  defaultsFunction = function defaultsFunction(element) {
```

Comment:
Not sure why this is here. It's not necessary to enclose the `for` loop in a function.